### PR TITLE
PROD-85: Add 'Simple form' + errors style + front-end validation

### PR DESCRIPTION
**Pull request by Pavliv**
# Reviewers

- [x] @vikabelous
- [x] @okohuch
- [x] @anastasiabalaban
- [x] @gnaizer 

https://ssu-jira.softserveinc.com/browse/PROD-85

# Description

Add gem Simple form
Add translations for fields and errors
Add front-end validation

# Checklist:

- [x] Rubocop doesn't raise any warnings and offenses
- [x] PR is approved by 2 team mates
- [x] PR is approved by 2 experts

### CODE REVIEW

File: config/locales/uk.yml [hunk 1]
```diff
@@ -172,20 +172,20 @@ uk:
...
+        other: занадто коротке (мінімум %{count} знаку)
       wrong_length:
-        few: неправильна довжина (має бути %{count} знаки)
-        many: неправильна довжина (має бути %{count} знаків)
-        one: неправильна довжина (має бути %{count} знак)
-        other: неправильна довжина (має бути %{count} знаку)
+        few: неправильної довжини (має бути %{count} знаки)
+        many: неправильної довжини (має бути %{count} знаків)
+        one: неправильної довжини (має бути %{count} знак)
+        other: неправильної довжини (має бути %{count} знаку)
```
Comment (gnaizer) [inline config/locales/uk.yml]: Please change "неправильної" to "невірної", also "має" to "повинно"

File: app/views/devise/registrations/new.html.haml [hunk 2]
```diff
@@ -1,34 +1,22 @@
...
-      = form_for(resource, as: resource_name, url: registration_path(resource_name)) do |f|
-        = devise_error_messages!
+    .col-lg-5.col-md-6
+      %h2.header= t('.sign_up')
+      = simple_form_for(resource, as: resource_name, url: registration_path(resource_name), html: { novalidate: false }) do |f|
         .form-group
-          = f.label :email
-          %br
-          = f.email_field :email, autofocus: true, autocomplete: "email", class: "form-control"
+          = f.input :email, label: 'Email', autofocus: true, autocomplete: 'email', input_html: {class: 'form-control'}, placeholder: 'example@mail.com', required: true
```
Comment (anastasiabalaban) [inline app/views/devise/registrations/new.html.haml]: add spaces in {}

File: app/views/devise/registrations/new.html.haml [hunk 3]
```diff
@@ -1,34 +1,22 @@
...
+          = f.input :password, autocomplete: 'off', input_html: { class: 'form-control', minlength: 6 }, label: false
         .form-group
-          = f.label :password_confirmation
-          %br
-          = f.password_field :password_confirmation, autocomplete: "off", class: "form-control"
+          = f.input :password_confirmation, autocomplete: 'off',  input_html: { class: 'form-control' }, required: true
         .actions
-          = f.submit t(".sign_up"), class: "btn btn-outline-info"
-      = render "devise/shared/links"
+          = f.submit t('.sign_up'), class: 'btn btn-outline-info'
```
Comment (anastasiabalaban) [inline app/views/devise/registrations/new.html.haml]: add class 'btn-size' . I add styles to buttons and we need to have the similar views.

File: app/views/devise/sessions/new.html.haml [hunk 4]
```diff
@@ -1,20 +1,16 @@
...
-          = f.password_field :password, autocomplete: "off", class: "form-control"
+          = f.input :password, autocomplete: 'off', input_html: {class: 'form-control'}
         - if devise_mapping.rememberable?
           .form-group
             = f.check_box :remember_me
             = f.label :remember_me
         .actions
-          = f.submit t(".sign_in"), class: "btn btn-outline-info"
-      = render "devise/shared/links"
+          = f.submit t('.sign_in'), class: 'btn btn-outline-info'
```
Comment (anastasiabalaban) [inline app/views/devise/sessions/new.html.haml]: add class 'btn-size'

Comment (Pavliv) [reply app/views/devise/sessions/new.html.haml]: added

File: app/views/devise/registrations/new.html.haml (hunk 3 above)
Comment (Pavliv) [reply app/views/devise/registrations/new.html.haml]: added

File: app/views/devise/registrations/new.html.haml (hunk 2 above)
Comment (Pavliv) [reply app/views/devise/registrations/new.html.haml]: added

File: config/locales/uk.yml (hunk 1 above)
Comment (Pavliv) [reply config/locales/uk.yml]: changed

**Review by anastasiabalaban — APPROVED** [review]

File: config/locales/uk.yml [hunk 5]
```diff
@@ -162,7 +162,7 @@ uk:
       greater_than: має бути більше ніж %{count}
       greater_than_or_equal_to: має бути більше ніж або дорівнювати %{count}
       inclusion: не включено до переліку
-      invalid: недійсний
+      invalid: введено не вірно
       less_than: має бути менше ніж %{count}
       less_than_or_equal_to: має бути менше ніж або дорівнювати %{count}
       model_invalid: 'Виникли помилки: %{errors}'
```
Comment (gnaizer) [inline config/locales/uk.yml]: same here with "має"

Comment (Pavliv) [reply config/locales/uk.yml]: I use only 'invalid: введено не вірно', line 165. I didn't use another lines.

**Review by gnaizer — APPROVED** [review]

File: app/views/devise/registrations/new.html.haml [hunk 6]
```diff
@@ -1,34 +1,22 @@
...
         .form-group
-          = f.label :surname, t(".surname")
-          %br
-          = f.text_field :surname, class: "form-control"
+          = f.input :surname, label: t('.surname'), input_html: { class: 'form-control', minlength: 3, maxlength: 30 }
         .form-group
-          = f.label :phone_number, t(".phone_number")
-          %br
-          = f.text_field :phone_number, class: "form-control"
+          = f.input :phone_number, label: t('.phone_number'), input_html: { class: 'form-control' }, placeholder: '+380XXXXXXXXX', required: false
```
Comment (okohuch) [inline app/views/devise/registrations/new.html.haml]: what if country is the USA? placeholder '+380XXXXXXXXX' is not correct

Comment (Pavliv) [reply app/views/devise/registrations/new.html.haml]: Edited

File: app/views/devise/registrations/new.html.haml [hunk 7]
```diff
@@ -1,34 +1,22 @@
...
         .form-group
-          = f.label :surname, t(".surname")
-          %br
-          = f.text_field :surname, class: "form-control"
+          = f.input :surname, label: t('.surname'), input_html: { class: 'form-control', minlength: 3, maxlength: 30 }
         .form-group
-          = f.label :phone_number, t(".phone_number")
-          %br
-          = f.text_field :phone_number, class: "form-control"
+          = f.input :phone_number, label: t('.phone_number'), input_html: { class: 'form-control' }, placeholder: '+XXXXXXXXXXXX', required: false
```
Comment (okohuch) [inline app/views/devise/registrations/new.html.haml]: but you agreed to remove a number completely, didn't you?
@shevchukm

Comment (shevchukm) [reply app/views/devise/registrations/new.html.haml]: yes, I did

Comment (okohuch) [reply app/views/devise/registrations/new.html.haml]: then please remove it everywhere

Comment (shevchukm) [reply app/views/devise/registrations/new.html.haml]: I've done

Comment (Pavliv) [reply app/views/devise/registrations/new.html.haml]: done

**Review by viktoriia-bilous — APPROVED** [review]

**Review by okohuch — APPROVED** [review]

**Pull request merged**